File: data-structures/records/c-sharp/player_record.cs
/*
Isaac Computer Science
Usage licensed under the Open Government Licence v3.0

Note: This file is designed to be copied out and compiled on your machine.
In order for it to compile properly you need to ensure that the project name is the same as the "namespace" in this file.
To run this file you need to:
1. Copy the contents
2. Paste them into the C# IDE of your choice (Visual Studio, for example)
3. Change the namespace to match your project (if necessary)
4. Compile the program
5. Run the program
*/

using System;

namespace IsaacCodeSamples
{
    // Use a class to represent a player as a record
    class PlayerRecord
    {
        public int playerNumber;
        public string firstName;
        public string lastName;
        public DateTime dateOfBirth;
        public string position;
        public bool injured;
    }


    class DataStructures
    {
        // The Main method is the entry point for all C# programs
        public static void Main()
        {
            // Create a player record and store the player details
            PlayerRecord player1 = new PlayerRecord();

            // Store the details of the player
            player1.playerNumber = 1;
            player1.firstName = "Maria";
            player1.lastName = "Oarps";
            player1.dateOfBirth = new DateTime(1994, 7, 3);
            player1.position = "Goalkeeper";
            player1.injured = false;

            // Display the player's name and position
            Console.WriteLine("### Player record ###");
            Console.WriteLine($"Name: {player1.firstName} {player1.lastName}");
            Console.WriteLine($"Position: {player1.position}");
        }

        
    }
}

File: data-structures/records/python/player_record.py
# Raspberry Pi Foundation
# Developed to be used alongside Isaac Computer Science,
# part of the National Centre for Computing Education
# Usage licensed under CC BY-SA 4


import datetime


class PlayerRecord():
    """Use a class to represent a player as a record"""
    
    def __init__ (self):
        self.player_number = None
        self.first_name = None
        self.last_name = None
        self.date_of_birth = None
        self.position = None
        self.injured = None


def main():
    """Create a player record and store the player details"""

    # Create a new player record
    player1 = PlayerRecord()

    # Store the details of the player
    player1.player_number = 1
    player1.first_name = "Maria"
    player1.last_name = "Oarps"
    player1.date_of_birth = datetime.datetime(1994, 7, 3)
    player1.position = "Goalkeeper"
    player1.injured = False

    # Display the player's name and position
    print("### Player record ###")
    print(f"Name: {player1.first_name} {player1.last_name}")
    print(f"Position: {player1.position}")


# This code will run if this file is executed directly
# (i.e. not called by another program)
if __name__ == '__main__':
    main()
